tests/app/questionnaire/test_placeholder_transforms.py

```diff
@@ -260,3 +260,141 @@ def test_telephone_number_link(self):
             self.transforms.telephone_number_link("012345 67890")
             == '<a href="tel:01234567890">012345 67890</a>'
         )
+
+    def generate_date_range(self, params):
+        with self.subTest():
+            for i, (
+                ref_date,
+                weeks_prior,
+                day_range,
+                first_day_of_week,
+                (date_from, date_to),
+            ) in enumerate(params):
+                expected = (date_from, date_to)
+                actual = self.transforms.generate_date_range(
+                    ref_date, weeks_prior, day_range, first_day_of_week
+                )
+                self.assertEqual(expected, actual, f"FAIL: Item {i}")
+
+    def test_generate_date_range_all_weekdays(self):
```

Comment:
I don't think we test the `lower()` method you added, so can we add some more (or refactor) test scenarios? Something like:
`("2021-09-27", -1, 7, "Monday", ("2021-09-20", "2021-09-26")),`
 ` ("2021-09-27", -2, 7, "MoNdAy", ("2021-09-13", "2021-09-19")),`